In [1]:
import geopandas as gp
import pandas as pd
import numpy as np
import math
import os
import pdv_functions as pdv

from functools import reduce

In [2]:
#Load in clean SOS file
elections_2022_raw = pd.read_excel("./raw-from-source/Election+Results+(11-08-2022).xlsx",sheet_name="Multi-Parish(Precinct)")

/Users/peterhorton/opt/anaconda3/envs/run_maup/lib/python3.8/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")


In [3]:
elections_2022_raw = elections_2022_raw[4:]

In [4]:
elections_2022_raw.reset_index(inplace = True, drop = True)

In [5]:
elections_2022_raw["Unnamed: 0"] = elections_2022_raw["Unnamed: 0"].fillna("NONE")

In [6]:
elections_2022_raw

Unnamed: 0 Louisiana Secretary of State                Unnamed: 2  \
0      U. S. Senator                          NaN                       NaN   
1               NONE     Beryl A. Billiot (NOPTY)  Gary Chambers, Jr. (DEM)   
2        Total Votes                         9378                    246933   
3             Acadia                          NaN                       NaN   
4              01 01                            5                        22   
...              ...                          ...                       ...   
44248        02 2010                           27                        11   
44249        02 2020                          122                        70   
44250        02 2030                           97                       104   
44251        02 2040                          143                       107   
44252  Early Voting                            20                        20   

                     Unnamed: 3          Unnamed: 4          Unnamed: 5  \
0                           NaN                 NaN                 NaN   
1      Devin Lance Graham (REP)  "Xan" John (OTHER)  John Kennedy (REP)   
2                         25275                2753              851568   
3                           NaN                 NaN                 NaN   
4                            13                   2                 456   
...                         ...                 ...                 ...   
44248                       NaN                 NaN                 NaN   
44249                       NaN                 NaN                 NaN   
44250                       NaN                 NaN                 NaN   
44251                       NaN                 NaN                 NaN   
44252                       NaN                 NaN                 NaN   

                                Unnamed: 6              Unnamed: 7  \
0                                      NaN                     NaN   
1      W. Thomas La Fontaine Olson (NOPTY)  Bradley McMorris (IND)   
2                                     1676                    5388   
3                                      NaN                     NaN   
4                                        0                       0   
...                                    ...                     ...   
44248                                  NaN                     NaN   
44249                                  NaN                     NaN   
44250                                  NaN                     NaN   
44251                                  NaN                     NaN   
44252                                  NaN                     NaN   

                     Unnamed: 8          Unnamed: 9  \
0                           NaN                 NaN   
1      MV "Vinny" Mendoza (DEM)  "Luke" Mixon (DEM)   
2                         11910              182887   
3                           NaN                 NaN   
4                             3                  24   
...                         ...                 ...   
44248                       NaN                 NaN   
44249                       NaN                 NaN   
44250                       NaN                 NaN   
44251                       NaN                 NaN   
44252                       NaN                 NaN   

                       Unnamed: 10            Unnamed: 11         Unnamed: 12  \
0                              NaN                    NaN                 NaN   
1      Salvador P. Rodriguez (DEM)  Aaron C. Sigler (LBT)  Syrita Steib (DEM)   
2                             7767                   4865               31568   
3                              NaN                    NaN                 NaN   
4                                0                      1                   0   
...                            ...                    ...                 ...   
44248                          NaN                    NaN                 NaN   
44249     

In [7]:
elections_2022_raw.loc[4131:4134]

Unnamed: 0  \
4131                                 Provisional Votes    
4132                                               NONE   
4133                                               NONE   
4134  U. S. Representative -- 1st Congressional Dist...   

     Louisiana Secretary of State Unnamed: 2 Unnamed: 3 Unnamed: 4 Unnamed: 5  \
4131                            0          0          0          0          1   
4132                          NaN        NaN        NaN        NaN        NaN   
4133                          NaN        NaN        NaN        NaN        NaN   
4134                          NaN        NaN        NaN        NaN        NaN   

     Unnamed: 6 Unnamed: 7 Unnamed: 8 Unnamed: 9 Unnamed: 10 Unnamed: 11  \
4131          0          0          0          1           0           0   
4132        NaN        NaN        NaN        NaN         NaN         NaN   
4133        NaN        NaN        NaN        NaN         NaN         NaN   
4134        NaN        NaN        NaN        NaN         NaN         NaN   

     Unnamed: 12 Unnamed: 13  
4131           0           0  
4132         NaN         NaN  
4133         NaN         NaN  
4134         NaN         NaN

In [8]:
slice_list = []
prev = 0
for val in range(1, len(elections_2022_raw)):
    if elections_2022_raw.loc[val, "Unnamed: 0"]=="NONE" and elections_2022_raw.loc[val+1, "Unnamed: 0"]=="NONE" :
        print(str(val))
        print(str(val+1))
        print(slice(val,val+1))
        #slice_list.append(slice(val,val+1))
        slice_list.append(slice(prev,val-1))
        prev = val+2

4132
4133
slice(4132, 4133, None)
4694
4695
slice(4694, 4695, None)
5361
5362
slice(5361, 5362, None)
5947
5948
slice(5947, 5948, None)
6877
6878
slice(6877, 6878, None)
7441
7442
slice(7441, 7442, None)
7852
7853
slice(7852, 7853, None)
8138
8139
slice(8138, 8139, None)
8388
8389
slice(8388, 8389, None)
8542
8543
slice(8542, 8543, None)
8956
8957
slice(8956, 8957, None)
9729
9730
slice(9729, 9730, None)
10676
10677
slice(10676, 10677, None)
10802
10803
slice(10802, 10803, None)
10955
10956
slice(10955, 10956, None)
11034
11035
slice(11034, 11035, None)
11164
11165
slice(11164, 11165, None)
11189
11190
slice(11189, 11190, None)
11326
11327
slice(11326, 11327, None)
11338
11339
slice(11338, 11339, None)
11359
11360
slice(11359, 11360, None)
11370
11371
slice(11370, 11371, None)
11382
11383
slice(11382, 11383, None)
11407
11408
slice(11407, 11408, None)
11419
11420
slice(11419, 11420, None)
11434
11435
slice(11434, 11435, None)
11451
11452
slice(11451, 11452, None)
11462
11463
slice(1146

In [9]:
interested_races = ['U. S. Senator',
'U. S. Representative -- 1st Congressional District',
'U. S. Representative -- 2nd Congressional District',
'U. S. Representative -- 3rd Congressional District',
'U. S. Representative -- 5th Congressional District',
'U. S. Representative -- 6th Congressional District',
'State Senator --  5th Senatorial District',
'State Senator -- 17th Senatorial District',
'CA No. 1 (ACT 130, 2021 - HB 154) -- Modifies the maximum amount of monies in certain state funds that may be invested in equities',
'CA No. 2 (ACT 172, 2022 - HB 599) -- Expands property tax exemptions for homestead exemption property for veterans with disabilities',
'CA No. 3 (ACT 156, 2021 - HB 315) -- Allows classified civil service employees to support election of family members to public office',
'CA No. 4 (ACT 155, 2021 - HB 59) -- Authorizes a political subdivision to waive charges for water under certain circumstances',
'CA No. 5 (ACT 133, 2021 - SB 154) -- Provides relative to property tax millage rate adjustments and maximum authorized millage rates',
'CA No. 6 (ACT 129, 2021 - HB 143) -- Limits the increase in assessed value of certain property following reappraisal in Orleans Parish',
'CA No. 7 (ACT 246, 2022 - HB 298) -- Provides relative to the prohibition of involuntary servitude and administration of criminal justice',
'CA No. 8 (ACT 171, 2022 - HB 395) -- Removes requirement of annual certification of income for certain eligible disabled homeowners',]

In [10]:
for slice_val in slice_list:
    subset = elections_2022_raw.loc[slice_val]
    subset.reset_index(inplace = True, drop = True)
    contest = subset.loc[0,"Unnamed: 0"]
    
    if contest in interested_races:
        #print(subset.loc[1])
        subset.columns = [str(i)+ "-:-" + contest for i in subset.loc[1] ]
        subset = subset.loc[2:]
        subset.reset_index(inplace = True, drop = True)
        #print(subset)
        
        county = "Statewide"
        county_dict = {}
        for val in range(0, len(subset[subset.columns[1]])):
            if math.isnan(subset.loc[val, subset.columns[1]]):
                #print(val)
                county = subset.loc[val,subset.columns[0]]
            county_dict[val] = county
        subset["COUNTY"] = subset.index.map(county_dict).fillna("BREAKPOINT")
        assert(subset[subset["COUNTY"]=="BREAKPOINT"].shape[0]==0)
        subset.rename(columns = {subset.columns[0]:"Precinct"}, inplace = True)        
        subset = subset[~subset[subset.columns[1]].isna()]
        
        subset.drop([i for i in subset.columns if i[:3] == "nan"], axis = 1, inplace = True)
        
        first_column = subset.pop('COUNTY')
        subset.insert(0, 'COUNTY', first_column)
        
        # Allocate results
        subset["UNIQUE_ID"] = subset["COUNTY"]+"-:-"+subset["Precinct"]
        race_cols = [i for i in subset.columns if i not in ["UNIQUE_ID","COUNTY","Precinct"]]
        #subset["PARISH"] = subset["UNIQUE_ID"].apply(lambda x: x.split("-:-")[0])
        statewide_vals = subset[subset["COUNTY"]=="Statewide"]
        print(statewide_vals.iloc[0])
        subset = subset[subset["COUNTY"]!="Statewide"]
        allocate_precs = subset[subset["UNIQUE_ID"].str.contains("Provisional") | subset["UNIQUE_ID"].str.contains("Early Voting")]
        receiving_votes = subset[~(subset["UNIQUE_ID"].str.contains("Provisional") | subset["UNIQUE_ID"].str.contains("Early Voting"))]
        
        subset_allocated = pdv.allocate_absentee(receiving_votes, allocate_precs, race_cols, "COUNTY", allocating_to_all_empty_precs=False)
        
        
        subset_allocated.to_csv("./cleaned_results/" + contest + ".csv", index = False)    

COUNTY                                                               Statewide
Precinct                                                           Total Votes
Beryl A. Billiot (NOPTY)-:-U. S. Senator                                  9378
Gary Chambers, Jr. (DEM)-:-U. S. Senator                                246933
Devin Lance Graham (REP)-:-U. S. Senator                                 25275
"Xan" John (OTHER)-:-U. S. Senator                                        2753
John Kennedy (REP)-:-U. S. Senator                                      851568
W. Thomas La Fontaine Olson (NOPTY)-:-U. S. Senator                       1676
Bradley McMorris (IND)-:-U. S. Senator                                    5388
MV "Vinny" Mendoza (DEM)-:-U. S. Senator                                 11910
"Luke" Mixon (DEM)-:-U. S. Senator                                      182887
Salvador P. Rodriguez (DEM)-:-U. S. Senator                               7767
Aaron C. Sigler (LBT)-:-U. S. Senator               

/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:270: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  precinct_specific_totals = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:352: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  first_allocation = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:374: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always sett

COUNTY                                                                                     Statewide
Precinct                                                                                 Total Votes
Katie Darling (DEM)-:-U. S. Representative -- 1st Congressional District                       61467
Howard Kearney (LBT)-:-U. S. Representative -- 1st Congressional District                       4907
Steve Scalise (REP)-:-U. S. Representative -- 1st Congressional District                      177670
UNIQUE_ID                                                                    Statewide-:-Total Votes
Name: 0, dtype: object


/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:352: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  first_allocation = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:374: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always setting a new array. To retain the old behavior, use either `df[df.columns[i]] = newvals` or, if columns are non-unique, `df.isetitem(i, newvals)`
  df_receiving_votes.loc[:,race]+=df_receiving_votes.loc[:,add_var]
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:375: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always 

COUNTY                                                                                     Statewide
Precinct                                                                                 Total Votes
Troy A. Carter (DEM)-:-U. S. Representative -- 2nd Congressional District                     158120
"Dan" Lux (REP)-:-U. S. Representative -- 2nd Congressional District                           46927
UNIQUE_ID                                                                    Statewide-:-Total Votes
Name: 0, dtype: object


/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:352: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  first_allocation = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:374: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always setting a new array. To retain the old behavior, use either `df[df.columns[i]] = newvals` or, if columns are non-unique, `df.isetitem(i, newvals)`
  df_receiving_votes.loc[:,race]+=df_receiving_votes.loc[:,add_var]
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:375: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always 

COUNTY                                                                                               Statewide
Precinct                                                                                           Total Votes
Clay Higgins (REP)-:-U. S. Representative -- 3rd Congressional District                                 144423
Holden Hoggatt (REP)-:-U. S. Representative -- 3rd Congressional District                                24474
Lessie Olivia Leblanc (DEM)-:-U. S. Representative -- 3rd Congressional District                         23641
Tia LeBrun (DEM)-:-U. S. Representative -- 3rd Congressional District                                    21172
Guy McLendon (LBT)-:-U. S. Representative -- 3rd Congressional District                                   1620
Thomas "Lane" Payne, Jr. (REP)-:-U. S. Representative -- 3rd Congressional District                       4012
Jacob "Jake" Shaheen (REP)-:-U. S. Representative -- 3rd Congressional District                           1955
G

/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:352: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  first_allocation = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:374: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always setting a new array. To retain the old behavior, use either `df[df.columns[i]] = newvals` or, if columns are non-unique, `df.isetitem(i, newvals)`
  df_receiving_votes.loc[:,race]+=df_receiving_votes.loc[:,add_var]
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:375: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always 

COUNTY                                                                                            Statewide
Precinct                                                                                        Total Votes
Oscar "Omar" Dantzler (DEM)-:-U. S. Representative -- 5th Congressional District                      35149
Allen Guillory (REP)-:-U. S. Representative -- 5th Congressional District                             12159
Walter Earl Huff (DEM)-:-U. S. Representative -- 5th Congressional District                           19383
Julia Letlow (REP)-:-U. S. Representative -- 5th Congressional District                              151080
Hunter Pullen (REP)-:-U. S. Representative -- 5th Congressional District                               5782
UNIQUE_ID                                                                           Statewide-:-Total Votes
Name: 0, dtype: object


/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:270: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  precinct_specific_totals = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:352: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  first_allocation = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:374: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always sett

COUNTY                                                                                            Statewide
Precinct                                                                                        Total Votes
Brian Belzer (REP)-:-U. S. Representative -- 6th Congressional District                               15535
Rufus Holt Craig, Jr. (LBT)-:-U. S. Representative -- 6th Congressional District                      30709
Garret Graves (REP)-:-U. S. Representative -- 6th Congressional District                             189684
UNIQUE_ID                                                                           Statewide-:-Total Votes
Name: 0, dtype: object


/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:352: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  first_allocation = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:374: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always setting a new array. To retain the old behavior, use either `df[df.columns[i]] = newvals` or, if columns are non-unique, `df.isetitem(i, newvals)`
  df_receiving_votes.loc[:,race]+=df_receiving_votes.loc[:,add_var]
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:375: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always 

COUNTY                                                                             Statewide
Precinct                                                                         Total Votes
Royce Duplessis (DEM)-:-State Senator --  5th Senatorial District                      17066
Mandie Landry (DEM)-:-State Senator --  5th Senatorial District                        15021
UNIQUE_ID                                                            Statewide-:-Total Votes
Name: 0, dtype: object
COUNTY                                                                                   Statewide
Precinct                                                                               Total Votes
Caleb Seth Kleinpeter (REP)-:-State Senator -- 17th Senatorial District                      22250
Jeremy S. LaCombe (DEM)-:-State Senator -- 17th Senatorial District                          18488
Kirk Rousset (REP)-:-State Senator -- 17th Senatorial District                                3370
UNIQUE_ID        

/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:270: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  precinct_specific_totals = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:352: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  first_allocation = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:374: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always sett

COUNTY                                                                                                                                                    Statewide
Precinct                                                                                                                                                Total Votes
YES-:-CA No. 1 (ACT 130, 2021 - HB 154) -- Modifies the maximum amount of monies in certain state funds that may be invested in equities                     475019
NO-:-CA No. 1 (ACT 130, 2021 - HB 154) -- Modifies the maximum amount of monies in certain state funds that may be invested in equities                      848366
UNIQUE_ID                                                                                                                                   Statewide-:-Total Votes
Name: 0, dtype: object


/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:270: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  precinct_specific_totals = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:352: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  first_allocation = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:374: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always sett

COUNTY                                                                                                                                                      Statewide
Precinct                                                                                                                                                  Total Votes
YES-:-CA No. 2 (ACT 172, 2022 - HB 599) -- Expands property tax exemptions for homestead exemption property for veterans with disabilities                     973937
NO-:-CA No. 2 (ACT 172, 2022 - HB 599) -- Expands property tax exemptions for homestead exemption property for veterans with disabilities                      361378
UNIQUE_ID                                                                                                                                     Statewide-:-Total Votes
Name: 0, dtype: object


/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:270: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  precinct_specific_totals = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:352: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  first_allocation = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:374: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always sett

COUNTY                                                                                                                                                       Statewide
Precinct                                                                                                                                                   Total Votes
YES-:-CA No. 3 (ACT 156, 2021 - HB 315) -- Allows classified civil service employees to support election of family members to public office                     434549
NO-:-CA No. 3 (ACT 156, 2021 - HB 315) -- Allows classified civil service employees to support election of family members to public office                      890055
UNIQUE_ID                                                                                                                                      Statewide-:-Total Votes
Name: 0, dtype: object


/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:270: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  precinct_specific_totals = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:352: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  first_allocation = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:374: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always sett

COUNTY                                                                                                                                               Statewide
Precinct                                                                                                                                           Total Votes
YES-:-CA No. 4 (ACT 155, 2021 - HB 59) -- Authorizes a political subdivision to waive charges for water under certain circumstances                     994361
NO-:-CA No. 4 (ACT 155, 2021 - HB 59) -- Authorizes a political subdivision to waive charges for water under certain circumstances                      332306
UNIQUE_ID                                                                                                                              Statewide-:-Total Votes
Name: 0, dtype: object


/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:352: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  first_allocation = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:374: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always setting a new array. To retain the old behavior, use either `df[df.columns[i]] = newvals` or, if columns are non-unique, `df.isetitem(i, newvals)`
  df_receiving_votes.loc[:,race]+=df_receiving_votes.loc[:,add_var]
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:375: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always 

COUNTY                                                                                                                                                      Statewide
Precinct                                                                                                                                                  Total Votes
YES-:-CA No. 5 (ACT 133, 2021 - SB 154) -- Provides relative to property tax millage rate adjustments and maximum authorized millage rates                     554257
NO-:-CA No. 5 (ACT 133, 2021 - SB 154) -- Provides relative to property tax millage rate adjustments and maximum authorized millage rates                      740006
UNIQUE_ID                                                                                                                                     Statewide-:-Total Votes
Name: 0, dtype: object


/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:352: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  first_allocation = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:374: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always setting a new array. To retain the old behavior, use either `df[df.columns[i]] = newvals` or, if columns are non-unique, `df.isetitem(i, newvals)`
  df_receiving_votes.loc[:,race]+=df_receiving_votes.loc[:,add_var]
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:375: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always 

COUNTY                                                                                                                                                        Statewide
Precinct                                                                                                                                                    Total Votes
YES-:-CA No. 6 (ACT 129, 2021 - HB 143) -- Limits the increase in assessed value of certain property following reappraisal in Orleans Parish                     636105
NO-:-CA No. 6 (ACT 129, 2021 - HB 143) -- Limits the increase in assessed value of certain property following reappraisal in Orleans Parish                      643076
UNIQUE_ID                                                                                                                                       Statewide-:-Total Votes
Name: 0, dtype: object


/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:270: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  precinct_specific_totals = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:352: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  first_allocation = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:374: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always sett

COUNTY                                                                                                                                                           Statewide
Precinct                                                                                                                                                       Total Votes
YES-:-CA No. 7 (ACT 246, 2022 - HB 298) -- Provides relative to the prohibition of involuntary servitude and administration of criminal justice                     508852
NO-:-CA No. 7 (ACT 246, 2022 - HB 298) -- Provides relative to the prohibition of involuntary servitude and administration of criminal justice                      790787
UNIQUE_ID                                                                                                                                          Statewide-:-Total Votes
Name: 0, dtype: object


/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:352: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  first_allocation = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:374: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always setting a new array. To retain the old behavior, use either `df[df.columns[i]] = newvals` or, if columns are non-unique, `df.isetitem(i, newvals)`
  df_receiving_votes.loc[:,race]+=df_receiving_votes.loc[:,add_var]
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:375: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always 

COUNTY                                                                                                                                                     Statewide
Precinct                                                                                                                                                 Total Votes
YES-:-CA No. 8 (ACT 171, 2022 - HB 395) -- Removes requirement of annual certification of income for certain eligible disabled homeowners                     717702
NO-:-CA No. 8 (ACT 171, 2022 - HB 395) -- Removes requirement of annual certification of income for certain eligible disabled homeowners                      593863
UNIQUE_ID                                                                                                                                    Statewide-:-Total Votes
Name: 0, dtype: object


/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:352: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  first_allocation = pd.DataFrame(df_receiving_votes.groupby([col_allocating]).sum())
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:374: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always setting a new array. To retain the old behavior, use either `df[df.columns[i]] = newvals` or, if columns are non-unique, `df.isetitem(i, newvals)`
  df_receiving_votes.loc[:,race]+=df_receiving_votes.loc[:,add_var]
/Users/peterhorton/Documents/RDH/pber/louisiana/2022/pdv_functions.py:375: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always 

In [11]:
path = "./cleaned_results/"
holder = []
for file in os.listdir(path):
    if file != ".DS_Store":
        temp = pd.read_csv(path+file)
        print(temp.columns)
        if file != "U. S. Senator.csv":
            temp.drop(["COUNTY", "Precinct"], axis = 1, inplace = True)
        else:
            temp.rename(columns = {"COUNTY":"PARISH"}, inplace = True)
        holder.append(temp)

Index(['COUNTY', 'Precinct',
       'YES-:-CA No. 6 (ACT 129, 2021 - HB 143) -- Limits the increase in assessed value of certain property following reappraisal in Orleans Parish',
       'NO-:-CA No. 6 (ACT 129, 2021 - HB 143) -- Limits the increase in assessed value of certain property following reappraisal in Orleans Parish',
       'UNIQUE_ID'],
      dtype='object')
Index(['COUNTY', 'Precinct',
       'YES-:-CA No. 1 (ACT 130, 2021 - HB 154) -- Modifies the maximum amount of monies in certain state funds that may be invested in equities',
       'NO-:-CA No. 1 (ACT 130, 2021 - HB 154) -- Modifies the maximum amount of monies in certain state funds that may be invested in equities',
       'UNIQUE_ID'],
      dtype='object')
Index(['COUNTY', 'Precinct',
       'YES-:-CA No. 4 (ACT 155, 2021 - HB 59) -- Authorizes a political subdivision to waive charges for water under certain circumstances',
       'NO-:-CA No. 4 (ACT 155, 2021 - HB 59) -- Authorizes a political subdivision to waiv

In [12]:
la_20_state_level = reduce(lambda x, y: pd.merge(x, y, on = "UNIQUE_ID", how = "outer"), holder)


In [13]:
first_column = la_20_state_level.pop('UNIQUE_ID')
la_20_state_level.insert(0, 'UNIQUE_ID', first_column)

In [14]:
la_20_state_level = la_20_state_level.fillna(0)

In [15]:
la_20_state_level

UNIQUE_ID  \
0     Acadia-:-01 01   
1     Acadia-:-01 02   
2     Acadia-:-01 03   
3     Acadia-:-01 04   
4     Acadia-:-01 05   
...              ...   
3932     Winn-:-07 1   
3933    Winn-:-07 1A   
3934     Winn-:-07 2   
3935    Winn-:-07 2A   
3936     Winn-:-07 7   

      YES-:-CA No. 6 (ACT 129, 2021 - HB 143) -- Limits the increase in assessed value of certain property following reappraisal in Orleans Parish  \
0                                                   221                                                                                              
1                                                   161                                                                                              
2                                                   166                                                                                              
3                                                   233                                                                                              
4                                                    81                                                                                              
...                                                 ...                                                                                              
3932                                                 28                                                                                              
3933                                                 31                                                                                              
3934                                                 61                                                                                              
3935                                                 45                                                                                              
3936                                                 40                                                                                              

      NO-:-CA No. 6 (ACT 129, 2021 - HB 143) -- Limits the increase in assessed value of certain property following reappraisal in Orleans Parish  \
0                                                   407                                                                                             
1                                                   377                                                                                             
2                                                   364                                                                                             
3                                                   435                                                                                             
4                                                   109                                                                                             
...                                                 ...                                                                                             
3932                                                 43                                                                                             
3933                                                 74                                                                                             
3934                                                 90                                                                                             
3935                                                117                                                                                             
3936                                                 97                                                                                             

      YES-:-CA No. 1 (ACT 130, 2021 - HB 154) -- Modifies the maximum amount of monies in certain state funds that may be invested i

In [16]:
def get_race(race_string):
    race_string = race_string.title()
    race_string = race_string.replace("(Vote For 1)","")
    if "U.S. House" in race_string or 'Us House' in race_string or "Representative To Congress" in race_string or "U. S. Representative" in race_string:
        return "CON"
    elif "State House" in race_string or "State Representative" in race_string:
        return "SL"
    elif "State Senate" in race_string or "State Senator" in race_string:
        return "SU"
    elif "President" in race_string:
        return "PRE"
    elif "US Senate" in race_string or "Us Senate" in race_string or "U.S. Senator" in race_string or "U. S. Senator" in race_string:
        return "USS"
    elif "Public Service" in race_string:
        return "PSC"
    elif "Attorney General" in race_string:
        return "ATG"
    elif "Auditor General" in race_string or "Auditor" in race_string:
        return "AUD"
    elif "Treasurer" in race_string:
        return "TRE"
    elif "Superintendent" in race_string:
        return "SUP"
    elif "Secretary Of State" in race_string:
        return "SOS"
    elif "Governor" in race_string:
        return "GOV"
    elif "Commissioner Of Labor" in race_string:
        return "LAB"
    elif "Commissioner Of Agriculture" in race_string:
        return "AGR"
    elif "Commissioner Of Insurance" in race_string:
        return "AGR"
    elif "Chief Justice Of The Supreme Court" in race_string:
        return "CJU"
    elif "Justice Of The Supreme Court" in race_string:
        return "JUS"
    elif "Ca No." in race_string:
        num = race_string.split(" ")[2]
        return "A"+str(num)

    else:
        print("No race for:", race_string)
        raise ValueError
        
def get_election_type(race_string):
    if "(runoff)" in race_string:
        return "R"
    else:
        return "G"
        
def get_party(race_string):
    if "(REP)" in race_string:
        return "R"
    elif "(DEM)" in race_string or "(Democrat)" in race_string:
        return "D"
    elif "(L)" in race_string or "(LBT)" in race_string:
        return "L"
    elif "(WI)" in race_string:
        return "O"
    elif "(IND)" in race_string:
        return "I"
    elif "(NOPTY)" in race_string:
        return "N"
    elif race_string[0:3]=="YES":
        return "YES"
    elif race_string[0:2]=="NO":
        return "NO"
    else:
        return "O"
           
def get_name(name_string):
    if "No." in name_string:
        return ""
    elif " (" in name_string and "Amendment" not in name_string and "Referendum" not in name_string:
        #print(name_string)
        name_string = name_string.split(" (")[0]
        name_string = name_string.replace("'","")
        likely_last = name_string.split(" ")[-1]
        proposed_last = likely_last[:3]
        if proposed_last in ['II', 'III', 'Jr', 'Jr.', 'Sr.', 'JR.', "JR", "IV"]:
            likely_last = name_string.split(" ")[-2]
            proposed_last = likely_last[:3]
        #print(proposed_last.upper())
        return proposed_last.upper()
    else:
        return name_string.split(" ")[-1][:3].upper()
import re
 

def get_district(race_string, fill_level):
    if "5th Senatorial District" in race_string:
        return "05"
    temp = race_string.split(" -- ")[1].split(" ")[0]
    return re.sub("\D","",temp).zfill(fill_level)

def column_rename_function(name_string):
    election_type = get_election_type(name_string)
    year = "22"
    party = get_party(name_string)
    race = get_race(name_string)
    district = ""
    #print(name_string)
    if race in ["CON", "SU"]:
        district = get_district(name_string, 2)
        year = ""
    elif race in ["SL"]:
        district = get_district(name_string, 3)
        year = ""
    name = get_name(name_string)
    new_col_name = election_type + year + race + district + party + name
#     print("election_type: ", election_type)
#     print("year: ", year)
#     print("race: ", race)
#     print("district: ", district)
#     print("party: ", party)
#     print("name: ", name)
    if len(new_col_name) > 10:
        print("LONG NAME")
        print(name_string, "->", new_col_name)
    return new_col_name

# Make a dictionary that points to the new column names and checks for duplicates

la_20_state_level.rename(columns = {"Lessie Olivia Leblanc (DEM)-:-U. S. Representative -- 3rd Congressional District":"Lessie OliviaLeblanc (DEM)-:-U. S. Representative -- 3rd Congressional District"}, inplace = True)

races_to_clean = [i for i in la_20_state_level.columns if i not in ["UNIQUE_ID", "PARISH", "Precinct"]]


race_updates_dict = {}
race_updates_reversed = {}
#clean_dups = {}
new_names = []
for val in races_to_clean:
    new_name = column_rename_function(val)
    race_updates_dict[val] = new_name
    if new_name not in new_names:
        new_names.append(new_name)
        race_updates_reversed[new_name] = val
    else:
        print("Duplicate", new_name)
        print(race_updates_reversed[new_name])
        print(val)
        #clean_dups[val] = race_updates_reversed[new_name]

In [17]:
race_updates_dict
pd.DataFrame(race_updates_dict.items()).to_csv("./field_names.csv", index = False)

In [18]:
la_20_state_level.rename(columns = race_updates_dict, inplace = True)

In [19]:
la_20_state_level

UNIQUE_ID  G22A6YES  G22A6NO  G22A1YES  G22A1NO  G22A4YES  G22A4NO  \
0     Acadia-:-01 01       221      407       152      496       446      204   
1     Acadia-:-01 02       161      377       174      377       363      189   
2     Acadia-:-01 03       166      364       120      426       376      165   
3     Acadia-:-01 04       233      435       188      498       483      197   
4     Acadia-:-01 05        81      109        65      133       128       68   
...              ...       ...      ...       ...      ...       ...      ...   
3932     Winn-:-07 1        28       43        19       53        49       24   
3933    Winn-:-07 1A        31       74        45       63        83       29   
3934     Winn-:-07 2        61       90        41      115       108       50   
3935    Winn-:-07 2A        45      117        43      119       112       54   
3936     Winn-:-07 7        40       97        30      108        95       48   

      G22A2YES  G22A2NO  GCON05DDAN  ...  G22A8YES  G22A8NO  GCON03RHIG  \
0          459      197         0.0  ...       304      344       568.0   
1          366      198         0.0  ...       242      310       417.0   
2          376      162         0.0  ...       223      317       439.0   
3          498      198         0.0  ...       335      310       532.0   
4          124       81         0.0  ...        93      106       103.0   
...        ...      ...         ...  ...       ...      ...         ...   
3932        41       29        33.0  ...        24       38         0.0   
3933        89       24         7.0  ...        56       51         0.0   
3934       124       34         0.0  ...       102       55         0.0   
3935       108       57        13.0  ...        78       90         0.0   
3936       102       40         3.0  ...        70       66         0.0   

      GCON03RHOG  GCON03DOLI  GCON03DLEB  GCON03LMCL  GCON03RPAY  GCON03RSHA  \
0           45.0        24.0        25.0         4.0        12.0         3.0   
1           66.0        37.0        34.0         8.0         9.0         3.0   
2           62.0        33.0        24.0         4.0         7.0         0.0   
3           86.0        43.0        26.0         5.0         9.0         3.0   
4           10.0        43.0        44.0         4.0         7.0         3.0   
...          ...         ...         ...         ...         ...         ...   
3932         0.0         0.0         0.0         0.0         0.0         0.0   
3933         0.0         0.0         0.0         0.0         0.0         0.0   
3934         0.0         0.0         0.0         0.0         0.0         0.0   
3935         0.0         0.0         0.0         0.0         0.0         0.0   
3936         0.0         0.0         0.0         0.0         0.0         0.0   

      GCON03IWIG  
0            6.0  
1           17.0  
2            7.0  
3           17.0  
4            7.0  
...          ...  
3932         0.0  
3933         0.0  
3934         0.0  
3935         0.0  
3936         0.0  

[3937 rows x 58 columns]

## Add County Fips

In [20]:
fips_file = pd.read_csv("./raw-from-source/FIPS/US_FIPS_Codes.csv")
fips_file = fips_file[fips_file["State"]=="Louisiana"]
fips_file["FIPS County"] = fips_file["FIPS County"].astype(str)
fips_file["FIPS County"] = fips_file["FIPS County"].str.zfill(3)
fips_file["County Name"] = fips_file["County Name"].replace("La Salle","Lasalle")
fips_dict = dict(zip(fips_file['County Name'], fips_file['FIPS County']))
la_20_state_level['COUNTYFP'] = la_20_state_level['PARISH'].str.replace("St.","St").map(fips_dict).fillna(la_20_state_level['PARISH'])
la_20_state_level['COUNTYFP'] = la_20_state_level['COUNTYFP'].astype(str).str.zfill(3)

# Print statements to check the county FIPs we've added
print(la_20_state_level['COUNTYFP'].unique())

['001' '003' '005' '007' '009' '011' '013' '015' '017' '019' '021' '023'
 '025' '027' '029' '031' '033' '035' '037' '039' '041' '043' '045' '047'
 '049' '051' '053' '055' '057' '059' '061' '063' '065' '067' '069' '071'
 '073' '075' '077' '079' '081' '083' '085' '087' '089' '091' '093' '095'
 '097' '099' '101' '103' '105' '107' '109' '111' '113' '115' '117' '119'
 '121' '123' '125' '127']


/var/folders/1t/0q4w6hm92mg_zxd84dfxmq3m0000gn/T/ipykernel_45263/1146220788.py:7: FutureWarning: The default value of regex will change from True to False in a future version.
  la_20_state_level['COUNTYFP'] = la_20_state_level['PARISH'].str.replace("St.","St").map(fips_dict).fillna(la_20_state_level['PARISH'])


In [21]:
race_cols = list(race_updates_dict.values())
race_cols.sort()

In [22]:
la_20_state_level["Precinct"] = la_20_state_level["UNIQUE_ID"].apply(lambda x: x.split("-:-")[1])

In [23]:
la_20_state_level.rename(columns = {"PARISH":"Parish"}, inplace = True)

In [24]:
la_20_state_level = la_20_state_level[["UNIQUE_ID", "COUNTYFP", 'Parish', "Precinct"]+ race_cols]

In [25]:
for race in race_cols:
    la_20_state_level.loc[:,race] = la_20_state_level.loc[:,race].astype(int)
    print(race, sum(la_20_state_level[race]))

G22A1NO 848366
G22A1YES 475019
G22A2NO 361378
G22A2YES 973937
G22A3NO 890055
G22A3YES 434549
G22A4NO 332306
G22A4YES 994361
G22A5NO 740006
G22A5YES 554257
G22A6NO 643076
G22A6YES 636105
G22A7NO 790787
G22A7YES 508852
G22A8NO 593863
G22A8YES 717702
G22USSDCHA 246933
G22USSDMEN 11910
G22USSDMIX 182887
G22USSDROD 7767
G22USSDSTE 31568
G22USSIMCM 5388
G22USSLSIG 4865
G22USSNBIL 9378
G22USSNOLS 1676
G22USSOJOH 2753
G22USSOWEN 1322
G22USSRGRA 25275
G22USSRKEN 851568
GCON01DDAR 61467
GCON01LKEA 4907
GCON01RSCA 177670
GCON02DCAR 158120
GCON02RLUX 46927
GCON03DLEB 21172
GCON03DOLI 23641
GCON03IWIG 3255
GCON03LMCL 1620
GCON03RHIG 144423
GCON03RHOG 24474
GCON03RPAY 4012
GCON03RSHA 1955
GCON05DDAN 35149
GCON05DHUF 19383
GCON05RGUI 12159
GCON05RLET 151080
GCON05RPUL 5782
GCON06LCRA 30709
GCON06RBEL 15535
GCON06RGRA 189684
GSU05DDUP 17066
GSU05DLAN 15021
GSU17DLAC 18488
GSU17RKLE 22250
GSU17RROU 3370


/var/folders/1t/0q4w6hm92mg_zxd84dfxmq3m0000gn/T/ipykernel_45263/829693378.py:2: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always setting a new array. To retain the old behavior, use either `df[df.columns[i]] = newvals` or, if columns are non-unique, `df.isetitem(i, newvals)`
  la_20_state_level.loc[:,race] = la_20_state_level.loc[:,race].astype(int)


Results on here: 
G22A1NO 848366
G22A1YES 475019
G22A2NO 361378
G22A2YES 973937
G22A3NO 890055
G22A3YES 434549
G22A4NO 332306
G22A4YES 994361
G22A5NO 740006
G22A5YES 554257
G22A6NO 643076
G22A6YES 636105
G22A7NO 790787
G22A7YES 508852
G22A8NO 593863
G22A8YES 717702
G22USSDCHA 246933
G22USSDMEN 11910
G22USSDMIX 182887
G22USSDROD 7767
G22USSDSTE 31568
G22USSIMCM 5388
G22USSLSIG 4865
G22USSNBIL 9378
G22USSNOLS 1676
G22USSOJOH 2753
G22USSOWEN 1322
G22USSRGRA 25275
G22USSRKEN 851568
GCON01DDAR 61467
GCON01LKEA 4907
GCON01RSCA 177670
GCON02DCAR 158120
GCON02RLUX 46927
GCON03DLEB 21172
GCON03DOLI 23641
GCON03IWIG 3255
GCON03LMCL 1620
GCON03RHIG 144423
GCON03RHOG 24474
GCON03RPAY 4012
GCON03RSHA 1955
GCON05DDAN 35149
GCON05DHUF 19383
GCON05RGUI 12159
GCON05RLET 151080
GCON05RPUL 5782
GCON06LCRA 30709
GCON06RBEL 15535
GCON06RGRA 189684
GSU05DDUP 17066
GSU05DLAN 15021
GSU17DLAC 18488
GSU17RKLE 22250
GSU17RROU 3370

In [26]:
la_20_state_level.columns

Index(['UNIQUE_ID', 'COUNTYFP', 'Parish', 'Precinct', 'G22A1NO', 'G22A1YES',
       'G22A2NO', 'G22A2YES', 'G22A3NO', 'G22A3YES', 'G22A4NO', 'G22A4YES',
       'G22A5NO', 'G22A5YES', 'G22A6NO', 'G22A6YES', 'G22A7NO', 'G22A7YES',
       'G22A8NO', 'G22A8YES', 'G22USSDCHA', 'G22USSDMEN', 'G22USSDMIX',
       'G22USSDROD', 'G22USSDSTE', 'G22USSIMCM', 'G22USSLSIG', 'G22USSNBIL',
       'G22USSNOLS', 'G22USSOJOH', 'G22USSOWEN', 'G22USSRGRA', 'G22USSRKEN',
       'GCON01DDAR', 'GCON01LKEA', 'GCON01RSCA', 'GCON02DCAR', 'GCON02RLUX',
       'GCON03DLEB', 'GCON03DOLI', 'GCON03IWIG', 'GCON03LMCL', 'GCON03RHIG',
       'GCON03RHOG', 'GCON03RPAY', 'GCON03RSHA', 'GCON05DDAN', 'GCON05DHUF',
       'GCON05RGUI', 'GCON05RLET', 'GCON05RPUL', 'GCON06LCRA', 'GCON06RBEL',
       'GCON06RGRA', 'GSU05DDUP', 'GSU05DLAN', 'GSU17DLAC', 'GSU17RKLE',
       'GSU17RROU'],
      dtype='object')

In [27]:
la_20_state_level.to_csv("./la_2022_gen_prec/la_2022_gen_prec.csv", index = False)